### PR TITLE
MM 44807 Adjust outline hover effects

### PR DESCRIPTION
**Pull request by m-ripper**
## Summary
The hover effects in the playbook outline are now more consistent.
Only section headers with a right header element are highlighted on hovering.

msedge_ZvIexJGS9g

## Ticket Link
[MM-44807](https://mattermost.atlassian.net/browse/MM-44807)

## Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

**Review by abhijit-singh — CHANGES_REQUESTED** [review]
Thanks @m-ripper! A couple of requests:
1. Clicking on the copy link icon on the left should not toggle the section on and off. Ideally, it should just copy the link and not trigger the toggle. Also, once you copy link, the hover state gets stuck.
2. Can we reduce the space between the header and the subtitle? The header section seems to have a margin-bottom of 10px currently which can be reduced to 2px.

**Comment by m-ripper** [comment]
> Thanks @m-ripper! A couple of requests:
> 
> 1. Clicking on the copy link icon on the left should not toggle the section on and off. Ideally, it should just copy the link and not trigger the toggle. Also, once you copy link, the hover state gets stuck.
> 2. Can we reduce the space between the header and the subtitle? The header section seems to have a margin-bottom of 10px currently which can be reduced to 2px.

Thanks @abhijit-singh for the feedback.

Regarding 2.: Should this affect all headers or only those with a subtitle?

**Comment by abhijit-singh** [comment]
> Regarding 2.: Should this affect all headers or only those with a subtitle?

Just the ones with a subtitle, which will be `Status updates` and `Retrospective` today.

File: webapp/src/components/backstage/playbook_editor/outline/section_retrospective.tsx [hunk 1]
```diff
@@ -61,7 +65,7 @@ const SectionRetrospective = ({playbook, refetch}: Props) => {
                 </BackstageSubheader>
                 <StyledSelect
                     value={retrospectiveReminderOptions.find((option) => option.value === playbook.retrospective_reminder_interval_seconds)}
-                    onChange={(option: {label: string; value: number;}) => {
+                    onChange={(option: { label: string; value: number; }) => {
```
Comment (iomodo) [inline webapp/src/components/backstage/playbook_editor/outline/section_retrospective.tsx:68]: extra spaces

File: webapp/src/components/backstage/playbook_editor/outline/outline.tsx [hunk 2]
```diff
@@ -88,6 +88,16 @@ const Outline = ({playbook, refetch}: Props) => {
                         />
                     </HoverMenuContainer>
                 )}
+                onHeaderClick={() => {
```
Comment (iomodo) [inline webapp/src/components/backstage/playbook_editor/outline/outline.tsx:92]: Could we create a single function to make sure that  toggle `onChange` and `onHeaderClick` does the same thing

File: webapp/src/components/backstage/playbook_editor/outline/outline.tsx [hunk 3]
```diff
@@ -122,6 +132,14 @@ const Outline = ({playbook, refetch}: Props) => {
...
```
Comment (iomodo) [inline webapp/src/components/backstage/playbook_editor/outline/outline.tsx:137]: same here

**Review by trilopin — APPROVED** [review]
🎉 LGTM

**Review by iomodo — APPROVED** [review]
Looks great @m-ripper, thanks <3

**Review by abhijit-singh — APPROVED** [review]

**Pull request merged**